feat: Add legal page with configuration options

aarondoet opened the pull request:
closes #591 

Allows you to enable a page with legal information which you can customize.

Only issue I have: I wanted to put the link in the footer - there is no footer in this project. Any ideas where to put the link instead? I was thinking of the home page, but that can be disabled and is not accessible for users that are logged in.

@ipod86 I guess you know a bit more about this, can you say if this would be enough to make it compliant with applicable laws (when a link to the page is placed somewhere)? Should something be changed? Should legal notice and privacy policy be separated or something like that?

![config](https://github.com/user-attachments/assets/11adf571-a1cd-4fbf-a67b-95d8ad730333)

![legal page](https://github.com/user-attachments/assets/5cf51215-cdce-4c85-b4ef-1857c0a86193)

aarondoet commented:
also again same as in #721, i changed the config, using `npx prisma db seed` updates it correctly but idk if a new migration is necessary nonetheless.

ipod86 commented:
Thank you for the good approach. 

To the questions. Data protection and imprint are two separate things that must be accessed separately. Strictly speaking, both must be accessible with their own link. 

The placement on the website is almost irrelevant. According to the law, it must be quick and easy to find and accessible with a maximum of two clicks. So it can be in a hamburger menu, for example. 

On implementation: 
I think it's good to be able to enter the text yourself. However, it would be very practical to have the choice between free text and a link. Our data protection declaration is a PDF with over 10 pages. You can also put a link to the imprint and data protection on the main website.

aarondoet commented:
Thanks for the input! I have updated it to have a text input and a url input. If a URL is provided the link (which I still have not placed anywhere yet) will be to that URL, otherwise it will show the provided content. I also split it up into imprint and privacy policy.

![config](https://github.com/user-attachments/assets/e7e59abe-4059-43f9-80ba-a902d05f65c2)

![imprint](https://github.com/user-attachments/assets/6732200a-cf75-46d0-8c87-92eb96df6104)

![privacy policy](https://github.com/user-attachments/assets/6a12e68e-a5d0-4a2f-8a3f-12a952f72b93)

![imprint](https://github.com/user-attachments/assets/6732200a-cf75-46d0-8c87-92eb96df6104)

ipod86 commented:
That looks very good. Great. Where can we see it afterwards on the main page?

stonith404 commented:
That's a really nice implementation, especially with markdown, thank you :) 

Regarding where to place the links, you could add a footer. In the past there was already a footer but I've removed it in this commit https://github.com/stonith404/pingvin-share/commit/c52a4d5e3ad717a10d15b7fe1dbf359b041c0976. You could add it back but replace the text "Made with 🖤  Elias Schneider" with "Powered by Pingvin Share" (or do you have a better idea?). And then place the two privacy policy and imprint links on the right side. 

What do you think about that?

Regarding your other PRs, I really appreciate it but I currently have much going on so it might take some time until I've reviewed them, sorry.

aarondoet commented:
Added a footer now. Has imprint and privacy policy on the left (if they exist), link to the project github io page in the middle (no i18n rn) and link to github repo on the right.

![footer](https://github.com/user-attachments/assets/34cb24f9-72cb-40e4-8065-ee66688cf60e)

ipod86 commented:
Test it immediately if it is in the official docker image.

aarondoet commented:
Would also close #335 I think

stonith404 reviewed (changes requested):
Great! I've noticed the following things that I would change:

1.  There is probably something wrong with the z-index of the footer because it's in front of the toasts and the backdrop of the dialogs:

Toasts:
![image](https://github.com/user-attachments/assets/2eb73b49-eb96-4243-8609-d75960963855)
Backdrop:
![image](https://github.com/user-attachments/assets/90ad8ebf-25d0-4b4b-945a-952b08fd7a6e)

2. I would remove the GitHub logo and directly link to the GitHub repository if you click "Pocket ID" in the footer because the page that you've linked is just the docs page.
3. I would mention in the description of the imprint and privacy policy that markdown is supported.

aarondoet commented:
oh the hint about markdown must have been lost when i separated privacy and imprint

stonith404 commented:
I've just improved the footer for mobile and now it can be merged. Thank you :)
